quark/Evaluator/pyeval.py

```diff
@@ -1,32 +1,65 @@
+# Thanks for the description of Dalvik bytecode instruction from the following websites, some of our explanations or
+# comments will quote from it.
+# https://source.android.google.cn/devices/tech/dalvik/instruction-formats
+# http://pallergabor.uw.hu/androidblog/dalvik_opcodes.html
+
+import logging
+from datetime import datetime
+
 from quark.Objects.TableObject import TableObject
 from quark.Objects.VarabileObject import VarabileObject
 
 MAX_REG_COUNT = 40
+TIMESTAMPS = datetime.now().strftime('%Y-%m-%d-%H-%M')
+LOG_FILENAME = f"{TIMESTAMPS}.quark.log"
+logging.basicConfig(level=logging.DEBUG, filename=LOG_FILENAME, filemode='w', datefmt='%Y-%m-%d %H:%M:%S')
+log = logging.getLogger(__name__)
+
+
+def logger(func):
+    def warp(*args, **kwargs):
+        log.info(f"{func.__name__} with args-> {args}")
+
+        func(*args, **kwargs)
+
+    return warp
 
 
 class PyEval:
     def __init__(self):
         # Main switch for executing the bytecode instruction.
         self.eval = {
+            # invoke-kind
             "invoke-virtual": self.INVOKE_VIRTUAL,
             "invoke-direct": self.INVOKE_DIRECT,
+            "invoke-static": self.INVOKE_STATIC,
+            # move-result-kind
+            "move-result": self.MOVE_RESULT,
+            "move-result-wide": self.MOVE_RESULT_WIDE,
             "move-result-object": self.MOVE_RESULT_OBJECT,
+            # instance-kind
             "new-instance": self.NEW_INSTANCE,
+            # const-kind
             "const-string": self.CONST_STRING,
+            "const": self.CONST,
             "const/4": self.CONST_FOUR,
+            "const/16": self.CONST_SIXTEEN,
+            "const/high16": self.CONST_HIGHSIXTEEN,
+            "const-wide": self.CONST_WIDE,
+            "const-wide/16": self.CONST_WIDE_SIXTEEN,
+            "const-wide/32": self.CONST_WIDE_THIRTY_TWO,
+            "const-wide/high16": self.CONST_WIDE_HIGHSIXTEEN,
+            # array
             "aget-object": self.AGET_OBJECT,
         }
 
         self.table_obj = TableObject(MAX_REG_COUNT)
         self.ret_stack = []
 
-    def __invoke(self, instruction):
+    def _invoke(self, instruction):
```

Comment:
At line 70 - 84, I think it can become only one loop.

Reply:
Thanks for your comments!

Reply:
@sidra-asa  When I tried to simplify this code, it leads to the unit test failed. I would leave this part unchanged until we start refactoring later, I still try to find the reason.

Reply:
I see. Leave this to next refactory